components/brave_perf_predictor/browser/perf_predictor_page_metrics_observer.cc

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_perf_predictor/browser/perf_predictor_page_metrics_observer.h"
+
+#include "brave/components/brave_perf_predictor/browser/perf_predictor_tab_helper.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/web_contents.h"
+
+namespace brave_perf_predictor {
+
+PerfPredictorPageMetricsObserver::PerfPredictorPageMetricsObserver() = default;
+
+PerfPredictorPageMetricsObserver::
+    ~PerfPredictorPageMetricsObserver() = default;
+
+page_load_metrics::PageLoadMetricsObserver::ObservePolicy
+PerfPredictorPageMetricsObserver::OnCommit(
+    content::NavigationHandle* navigation_handle,
+    ukm::SourceId source_id) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
```

Comment:
what is the sequence checker here for?

Reply:
intended to check that the observer methods are called in a correct sequence (no first paint before navigation is committed), some of the built-in metric observers in chrome/browser/page_load_metrics/observers use that, but it shouldn't be strictly necessary

Reply:
it doesn't check to make sure they are called in the "correct" sequence, it checks to make sure they are in sequence and it does that by making sure they are all called on the same thread. Not sure why they have them in some of the observers, there's really no reason that I can see. You typically use them when you have a class where a subset of methods are called on a different thread

Reply:
it doesn't hurt anything, just not a typical usage

Reply:
but just to be clear there is no ordering aspect to it, it's purely for thread safety

Reply:
ok, if it doesn't do anything useful either, might as well remove them